cpp/src/ice2slice/Gen.cpp

```diff
@@ -153,10 +197,13 @@ namespace
         ContainedPtr contained = dynamic_pointer_cast<Contained>(type);
         if (contained)
         {
-            bool hasCSharpNamespaceAttribute;
-            string csharpNamespace = getCSharpNamespace(contained, hasCSharpNamespaceAttribute);
-
-            os << csharpNamespace << "." << contained->name();
+            string csNamespacePrefix = getCsNamespacePrefix(contained);
+            string scoped = getCsMappedScoped(contained, ".").substr(1);
+            if (!csNamespacePrefix.empty())
+            {
+                os << csNamespacePrefix << ".";
+            }
+            os << scoped;
```

Comment:
I think these should be equivalent:
```suggestion
            string csNamespace = getCsMappedNamespace(contained);
            os << csNamespace << "." << getCsMappedName();
```